tree.go

```diff
@@ -218,11 +219,13 @@ func (t *Tree) IterateRangeInclusive(start, end []byte, ascending bool, fn func(
 	})
 }
 
-// Clone creates a clone of the tree. Used internally by VersionedTree.
+// Clone creates a clone of the tree and increments its version.
+// Used internally by VersionedTree.
 func (tree *Tree) clone() *Tree {
 	return &Tree{
-		root: tree.root,
-		ndb:  tree.ndb,
+		root:    tree.root,
+		ndb:     tree.ndb,
+		version: tree.version + 1,
```

Comment:
update on clone???
not after SaveVersion?

I would suggest starting at 1 on init, and increment after SaveVersion, seems cleaner...
Expecially, as we want to clone Tree for caching, etc.

Reply:
That clone method is private and only used by VersionedTree, but I hear you - will change.

Reply:
9d1aa62